### PR TITLE
resolve server listen details with server.address()

### PR DESCRIPTION
**Pull request by btkostner**
This should work (made right on GitHub without testing :grin:).

### Issue:
When using port 0, linux assigns a random unused port to the server. Right now this will be outputted as `localhost:0` and you will have no clue what the actual port is. Usually this happens when doing some CI testing and you want to avoid port collisions.

### Fix:
This uses the `server.address()` method to resolve the real host and port the server is listening on.

### Node docs:
https://nodejs.org/api/net.html#net_server_address

### CODE REVIEW

File: lib/core/nuxt.js [hunk 1]
```diff
@@ -129,7 +129,13 @@ export default class Nuxt {
             return reject(err)
           }
 
-          const listenURL = chalk.underline.blue(`http://${host}:${port}`)
+          const connection = server.address()
+          const realHost = (['127.0.0.1', '0.0.0.0'].includes(connection.address))
+            ? 'localhost'
+            : connection.address
+          const realPort = server.address().port
```
Comment (clarkdo) [inline lib/core/nuxt.js]: Should be connection.port ?

File: lib/core/nuxt.js [hunk 2]
```diff
@@ -129,7 +129,13 @@ export default class Nuxt {
             return reject(err)
           }
 
-          const listenURL = chalk.underline.blue(`http://${host}:${port}`)
+          const connection = server.address()
```
Comment (clarkdo) [inline lib/core/nuxt.js]: How about  const { address: host, port }= server.address()

File: lib/core/nuxt.js (hunk 1 above)
Comment (btkostner) [reply lib/core/nuxt.js]: Yes it should

File: lib/core/nuxt.js (hunk 2 above)
Comment (btkostner) [reply lib/core/nuxt.js]: `port` was already defined and I wasn't sure what the nuxt stance on nesting variables was. I'll update it.

File: lib/core/nuxt.js [hunk 3]
```diff
@@ -129,6 +129,11 @@ export default class Nuxt {
             return reject(err)
           }
 
+          let { host, port } = server.address()
```
Comment (clarkdo) [inline lib/core/nuxt.js]: Should be:

```js
  ({ host, port } = server.address())
```

**Review by clarkdo — APPROVED** [review]
LGTM

**Comment by Atinux** [comment]
Great work @btkostner 

Thanks @clarkdo for your review

**Pull request merged**